chiron/mcmc.py

```diff
@@ -112,103 +175,802 @@ def run(
+        """
+        Implement the logic specific to displacement changes.
+
+        Parameters
+        ----------
+        current_sampler_state : SamplerState, required
+            Current sampler state.
+        current_thermodynamic_state : ThermodynamicState, required
+            Current thermodynamic state.
+        current_reduced_pot : float, required
+            Current reduced potential.
+        current_nbr_list : PairsBase, optional
+            Neighbor list associated with the current state.
+
+        Returns
+        -------
+        proposed_sampler_state : SamplerState
+            Proposed sampler state.
+        proposed_thermodynamic_state : ThermodynamicState
+            Proposed thermodynamic state.
+        proposed_reduced_pot : float
+            Proposed reduced potential.
+        log_proposal_ratio : float
+            Log proposal ratio.
+        proposed_nbr_list : PairsBase
+            Proposed neighbor list. If not defined, this will be None.
+
+        """
+        from loguru import logger as log
+
+        key = current_sampler_state.new_PRNG_key
+
+        import jax.random as jrandom
+
+        nr_of_atoms = current_sampler_state.n_particles
+
+        initial_volume = (
+            current_sampler_state.box_vectors[0][0]
+            * current_sampler_state.box_vectors[1][1]
+            * current_sampler_state.box_vectors[2][2]
+        )
+
+        # Calculate the maximum amount the volume can change by
+        delta_volume_max = self.volume_max_scale * initial_volume
+
+        # Calculate the volume change by generating a random number between -1 and 1
+        # and multiplying by the maximum allowed volume change, delta_volume_max
+        delta_volume = jrandom.uniform(key, minval=-1, maxval=1) * delta_volume_max
+        # calculate the new volume
+        proposed_volume = initial_volume + delta_volume
+
+        # calculate the length scale factor for particle positions and box vectors
+        length_scaling_factor = jnp.power(proposed_volume / initial_volume, 1.0 / 3.0)
+
+        import copy
+
+        proposed_sampler_state = copy.deepcopy(current_sampler_state)
+        proposed_sampler_state.x0 = current_sampler_state.x0 * length_scaling_factor
+
+        proposed_sampler_state.box_vectors = (
+            current_sampler_state.box_vectors * length_scaling_factor
+        )
+
+        if current_nbr_list is not None:
+            proposed_nbr_list = copy.deepcopy(current_nbr_list)
+            # after scaling the box vectors and coordinates we should always rebuild the neighborlist
+            proposed_nbr_list.build(
+                proposed_sampler_state.x0, proposed_sampler_state.box_vectors
+            )
+
+        proposed_reduced_pot = current_thermodynamic_state.get_reduced_potential(
+            proposed_sampler_state, proposed_nbr_list
+        )
+
+        #  ⎡−β (ΔU + PΔV ) + N ln(V new /V old )⎤
```

Comment:
we might want to give a reference for this, e.g.
eq.33, http://dx.doi.org/10.1155/2015/183918

Reply:
Or we could go back to the original source: https://www.tandfonline.com/doi/abs/10.1080/00268977200100031